src/Compiler/Postprocessor.php

```diff
@@ -30,26 +37,61 @@ final class Postprocessor
 
     private CompilerInterface $compiler;
 
+    private ScopeFactory $scopeFactory;
+
+    private NodeScopeResolver $nodeScopeResolver;
+
     public function __construct(
         Parser $parser,
         NodeVisitorStorage $nodeVisitorStorage,
         Standard $printerStandard,
-        CompilerInterface $compiler
+        CompilerInterface $compiler,
+        ScopeFactory $scopeFactory,
+        NodeScopeResolver $nodeScopeResolver
     ) {
         $this->parser = $parser;
         $this->nodeVisitorStorage = $nodeVisitorStorage;
         $this->printerStandard = $printerStandard;
         $this->compiler = $compiler;
+        $this->scopeFactory = $scopeFactory;
+        $this->nodeScopeResolver = $nodeScopeResolver;
     }
 
-    public function postProcess(string $phpContent, Template $template): string
+    public function postProcess(string $phpContent, Template $template, string $compileFilePath): string
     {
         $phpStmts = $this->findNodes($phpContent);
         foreach ($this->nodeVisitorStorage->getNodeVisitors() as $nodeVisitors) {
             $phpStmts = $this->processNodeVisitors($phpStmts, $nodeVisitors, $template);
         }
 
-        return $this->printerStandard->prettyPrintFile($phpStmts);
+        $phpContent = $this->printerStandard->prettyPrintFile($phpStmts);
+        file_put_contents($compileFilePath, $phpContent);
```

Comment:
Why require is needed here? Would not there be a conflict if version required here and final version is different?

Reply:
Phpstan needs to know all symbols. Final version has to have the same classes as the version at this place